mlflow/server/handlers.py

```diff
@@ -262,6 +262,44 @@ def wrapper(*args, **kwargs):
 ]
 
 
+def _disable_unless_serve_artifacts(func):
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        from mlflow.server import SERVE_ARTIFACTS_ENV_VAR
+
+        if not os.environ.get(SERVE_ARTIFACTS_ENV_VAR):
+            return Response(
+                (
+                    f"Endpoint: {request.url_rule} disabled due to the mlflow server running "
+                    "without `--serve-artifacts`. To enable artifacts server functionaltiy, "
+                    "run `mlflow server` with `--serve-artfiacts`"
+                ),
+                503,
+            )
+        return func(*args, **kwargs)
+
+    return wrapper
+
+
+def _disable_if_artifacts_only(func):
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        from mlflow.server import ARTIFACTS_ONLY_ENV_VAR
+
+        if os.environ.get(ARTIFACTS_ONLY_ENV_VAR):
+            return Response(
+                (
+                    "Endpoint disabled due to the mlflow server running in `--artifacts-only` "
```

Comment:
```suggestion
                    f"Endpoint {request.url_rule} disabled due to the mlflow server running in `--artifacts-only` "
```

Can we also include `request.url_rule` here?